src/platform_impl/linux/wayland/window/shim.rs

```diff
@@ -188,6 +214,52 @@ impl WindowHandle {
         }
     }
 
+    pub fn set_user_attention(&self, request_type: Option<UserAttentionType>) {
+        let xdg_activation = match self.xdg_activation.as_ref() {
+            None => return,
+            Some(xdg_activation) => xdg_activation,
+        };
+
+        // Unsetting is done by compositor.
+        if request_type.is_none() {
+            return;
+        }
+
+        // The token is already present, nothing to set.
+        if self.xdg_activation_token.borrow().is_some() {
+            return;
+        }
```

Comment:
Is this the only reason why the `xdg_activation_token` needs to be stored? From a Rust point of view, I don't see much other use for it, but maybe there's some destructor stuff that requires keeping it alive?

Reply:
It's getting destroyed via destroy method in a callback, so it's stored just to be properly cleaned.

Reply:
But what is destroyed is the token that is passed in the callback, not the token stored, right? So the storage servers no purpose?

Reply:
They are the same, but I get what you're talking about. I think we can use `Cell<bool>` for that.

Reply:
Yeah, exactly. If the goal is only to not activate multiple times, a boolean is sufficient. Atomic might work?

Reply:
Nah, you don't need atomic here it's single threaded in all cases since queue isn't sync + send.

Reply:
Updated.